src/CloudSearchDomain/CloudSearchDomainClient.php

```diff
@@ -35,6 +36,7 @@ public static function getArguments()
             // (e.g. http://search-blah.{region}.cloudsearch.amazonaws.com)
             return explode('.', new Uri($args['endpoint']))[1];
         };
+        unset($args['endpoint']['default']);
```

Comment:
Why is this necessary?  This seems like an endpoint is always required, so it should never go to the default?

Reply:
The client resolver won't handle the 'required' property if there's a default